lxd/instance/drivers/driver_lxc.go

```diff
@@ -6232,7 +6229,57 @@ func (d *lxc) insertMountLXC(source, target, fstype string, flags int) error {
 	return nil
 }
 
+func (d *lxc) insertMountElegant(source, target, fstype string, flags int, idmapType idmap.IdmapStorageType) error {
```

Comment:
Please can we have this function accept a `ctx context.Context` argument that can be passed to `RunCommandInheritFds` rather than `context.TODO()` to enable cancellation if needed by the caller?

Reply:
Yeah, I have no idea about that TODO() stuff. That seems to have been added as a hack after I wrote that code?

Reply:
`context.TODO()` is part of the standard library, and it allows passing a context to something that requires it without needing to think about the cancellation scenarios at that time. So it allows for transitioning functions over to context. Not really a hack.

If however you feel that this particular function shouldn't be cancellable, then its also fine to not accept a context, and then instead pass `context.Background()` to the `RunCommandInheritFds` so that it uses a context that never gets cancelled.

The `context.TODO()` is a warning sign that something should be changed here in the future.
So I think it would be better to either delegate that decision to the caller, by way of a `ctx` argument or use `context.Background()` inside the function.

Reply:
I'm sorry, that TODO() stuff is strewn all across that file and I'm not sure how that works. So this should probably be a follow up commit where all those functions are converted accordingly.

Reply:
Would it be safe for Go to kill this command in the case of the caller needing to cancel it? If not then we should use context.Background() here rather than context.TODO(). If its safe to cancel, then I would leave it as is, and we can revisit later.

Reply:
"Safe" as in "isn't a security issue? Probably, yes. "Safe" as in "wouldn't leave the container with wonky mounts" then probably not. :)

Reply:
Yeah I was thinking from the wonky mounts perspective, in that case I'd suggest using context.Background() to make it explicit it you don't want this to be later converted to a cancellable context (e.g. its not TODO).

Reply:
Ok, done! Thanks for the help with that.